sphinx/domains/std.py

```diff
@@ -202,7 +204,19 @@ def handle_signature(self, sig: str, signode: desc_signature) -> str:
             if count:
                 signode += addnodes.desc_addname(', ', ', ')
             signode += addnodes.desc_name(optname, optname)
-            signode += addnodes.desc_addname(args, args)
+            if self.env.config.option_parse_variable_part:
+                for part in samp_role.parse(args):
+                    if isinstance(part, nodes.Text):
+                        text = part.astext()
+                        if not text.strip():
+                            signode += addnodes.desc_sig_space()
+                        else:
+                            # in order to preserve whitespaces, call desc_addname
+                            signode += addnodes.desc_addname(text, text)
```

Comment:
Hmm, is ``desc_addname`` appropriate? In the "normal" programming language domains it is used for additional name parts, i.e., qualification of the main declared entity. But here it is not really for typesetting any name part.
Now that we are anyway changing this: to be conservative I suggest using the original ``nodes.Text`` until something a more specific markup is deemed appropriate.

Reply:
Hm, then I would need to find a way how to preserve whitespaces for a `nodes.Text` node.
E.g. for:

```
.. option:: -o directory, --object-directory directory, --object-file file
```

parts are: `_directory`, `_directory`, `_file`, where `_` is a space. If I don't do it, I would end up with:
![Screenshot from 2022-03-23 10-36-23](https://user-images.githubusercontent.com/2658545/159668708-9623fb51-f573-488e-ad74-2520a1873034.png)

Reply:
How about these changes: https://github.com/jakobandersen/sphinx/tree/marxin-option_parse_variable_part